atc/worker/gardenruntime/worker.go

```diff
@@ -713,3 +640,97 @@ func (worker *Worker) GardenClient() gclient.Client {
 func (worker *Worker) BaggageclaimClient() baggageclaim.Client {
 	return worker.bcClient
 }
+
+func (worker *Worker) acquireVolumeStreamingLock(
+	ctx context.Context,
+	resourceCacheID int,
+	destination string,
+) (lock.Lock, bool, error) {
+	// if caching of streamed volumes is not enabled then we should not globally
+	// lock the volume streaming since steps cannot share the streamed volumes as caches
+	if !atc.EnableCacheStreamedVolumes {
```

Comment:
Say a `get` is followed by a `in_parallel`, and `in_parallel` has 3 `task`s that all take the `get` resource as input. When the 3 tasks happen to place on the same worker, it will initiate 3 volume streams. In that case, should this lock also help?

Reply:
If `EnableCacheStreamedVolumes` is not enabled then they should all stream since the volume streamed for the first task isn't a valid cache for the second task. In the case where `EnableCacheStreamedVolumes` is enabled then the lock should kick in and only one of those tasks will actually do the streaming and the other 2 will wait.